pkg/server/cache/entrycache/fullcache.go

```diff
@@ -173,6 +178,37 @@ func Build(ctx context.Context, entryIter EntryIterator, agentIter AgentIterator
 	}, nil
 }
 
+func Update(ctx context.Context, ds datastore.DataStore, cache *FullEntryCache) error {
+	req := &datastore.ListEventsRequest{
+		LastID: cache.lastEventID,
+	}
+	resp, err := ds.ListEvents(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	for _, entryID := range resp.EntryIDs {
```

Comment:
based on datastore layer, we may get repeated entries IDs
so this block can be doing the same search multiple times.